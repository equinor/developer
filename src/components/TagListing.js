import { Link } from "gatsby";
import React from "react";
import styled from "styled-components";
import kebabCase from "lodash/kebabCase";
import Style from "../ui/style";
import { Actions } from "../reducers/SearchReducer";

const Wrapper = styled.section`
  display: flex;
  margin-left: 15px;
  margin-right: 15px;
  margin: 0 15px 40px;
`;

const Title = styled.span`
  margin-top: 10px;
  display: inline-flex;
  font-size: ${Style.typography.base};
  font-weight: bold;
  line-height: 21px;
  flex-wrap: wrap;
  margin-right: 15px;
`;

const List = styled.ul`
  margin-top: 40px;
  list-style: none;
  display: inline-flex;
  flex-wrap: wrap;
  padding-left: 20px;
`;

const ListItem = styled.li`
  background-color: ${props =>
    props.disabled ? Style.colors.lighterGray : Style.colors.lichenGreen};
  padding: 0 20px;
  text-align: center;
  border-radius: 20.5px;
  margin-right: 20px;
  margin-bottom: 10px;

  a,
  span {
    font-size: 12px;
    font-weight: 500;
    line-height: 14px;
    color: ${Style.colors.mossGreen};
    text-transform: uppercase;
  }
`;

const FilterListItem = styled(ListItem)`
  cursor: pointer;
`;

const TagWrapper = props => {
  return (
    <Wrapper>
      <List>
        <Title>Tags</Title>
        {props.children}
      </List>
    </Wrapper>
  );
};

/**
 * @param selectedTags InitialSelectedTags
 * @returns {function({node: *}): boolean}
 */
export function filterTags(selectedTags) {
  return ({ node }) => {
    const tags = node.frontmatter.tags.filter(tag => {
      const isDisabled = selectedTags[tag.toUpperCase()];
      return !isDisabled;
    });
    return tags.length > 0;
  };
}

export const TagFilter = ({ tags, selectedTags, dispatch }) => {
  const handleSelectedTags = value =>
    dispatch({ type: Actions.TOGGLE_SELECTED_TAG, value });
  const onSelectAll = () => dispatch({ type: Actions.SELECT_ALL_TAGS });

  const TagStatus = styled.span`
    font-size: 24px;
    padding-left: 10px;
    fontweight: 500;
  `;

  return (
    <TagWrapper>
      {tags.map(tag => {
        const handleClick = () => handleSelectedTags(tag.fieldValue);
        const disabled = selectedTags[tag.fieldValue.toUpperCase()];
        return (
          <FilterListItem
            disabled={disabled}
            key={tag.fieldValue}
            onClick={handleClick}
          >
            <span>
              {tag.fieldValue}
              <TagStatus>{disabled ? "+" : "x"}</TagStatus>
            </span>
          </FilterListItem>
        );
      })}
      <span
        style={{ textDecoration: "underline", cursor: "pointer" }}
        onClick={onSelectAll}
      >
        Select all
      </span>
    </TagWrapper>
  );
};

export default ({ tags }) => {
  return (
    <TagWrapper>
      {tags.map(tag => (
        <ListItem key={tag.fieldValue}>
          <Link to={`/tags/${kebabCase(tag.fieldValue)}/`}>
            {tag.fieldValue}
          </Link>
        </ListItem>
      ))}
    </TagWrapper>
  );
};
