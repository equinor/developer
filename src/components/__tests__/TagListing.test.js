import React from "react";
import renderer from "react-test-renderer";
import TagListing from "../TagListing";

describe("BlogListing", () => {
  it("renders correctly", () => {
    const tags = [{ fieldValue: "Api" }, { fieldValue: "Tech", enabled: true }];
    const tree = renderer.create(<TagListing tags={tags} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
