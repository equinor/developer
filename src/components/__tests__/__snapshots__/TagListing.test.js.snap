// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogListing renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: 15px;
  margin-right: 15px;
  margin: 0 15px 40px;
}

.c2 {
  margin-top: 10px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 16px;
  font-weight: bold;
  line-height: 21px;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-right: 15px;
}

.c1 {
  margin-top: 40px;
  list-style: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding-left: 20px;
}

.c3 {
  background-color: #EAEAEA;
  padding: 0 20px;
  text-align: center;
  border-radius: 20.5px;
  margin-right: 20px;
  margin-bottom: 10px;
}

.c3 a,
.c3 span {
  font-size: 12px;
  font-weight: 500;
  line-height: 14px;
  color: #007079;
  text-transform: uppercase;
}

.c4 {
  background-color: #DEFAEB;
  padding: 0 20px;
  text-align: center;
  border-radius: 20.5px;
  margin-right: 20px;
  margin-bottom: 10px;
}

.c4 a,
.c4 span {
  font-size: 12px;
  font-weight: 500;
  line-height: 14px;
  color: #007079;
  text-transform: uppercase;
}

<section
  className="c0"
>
  <ul
    className="c1"
  >
    <span
      className="c2"
    >
      Tags
    </span>
    <li
      className="c3"
    >
      <a
        href="/tags/api/"
      >
        Api
      </a>
    </li>
    <li
      className="c4"
    >
      <a
        href="/tags/tech/"
      >
        Tech
      </a>
    </li>
  </ul>
</section>
`;
